addons/website/static/src/snippets/s_image_gallery/gallery_slider.js

```diff
@@ -16,121 +16,167 @@ export class GallerySlider extends Interaction {
     };
 
     setup() {
+        // Compatibility with older versions of .o_slideshow
+        this.snippetVersion = Number(this.el.dataset.vcss);
         this.hideOnClickIndicator = true;
-        this.carouselEl = this.el.classList.contains("carousel") ? this.el : this.el.querySelector(".carousel");
+
+        this.carouselEl = this.el.classList.contains("carousel")
+            ? this.el
+            : this.el.querySelector(".carousel");
         this.indicatorEl = this.carouselEl?.querySelector(".carousel-indicators");
         if (this.indicatorEl) {
-            this.prevEl = this.indicatorEl.querySelector("li.o_indicators_left");
-            this.nextEl = this.indicatorEl.querySelector("li.o_indicators_right");
-            if (this.prevEl) {
-                this.prevEl.style.visibility = ""; // force visibility as some databases have it hidden
-            }
-            if (this.nextEl) {
-                this.nextEl.style.visibility = "";
-            }
-            this.liEls = this.indicatorEl.querySelectorAll("li[data-bs-slide-to]");
-            let indicatorWidth = this.indicatorEl.getBoundingClientRect().width;
-            if (indicatorWidth === 0) {
-                // An ancestor may be hidden so we try to find it and make it
-                // visible just to take the correct width.
-                let indicatorParentEl = this.indicatorEl.parentElement;
-                while (indicatorParentEl) {
-                    if (!isVisible(indicatorParentEl)) {
-                        if (!indicatorParentEl.style.display) {
-                            indicatorParentEl.style.display = "block";
-                            indicatorWidth = this.indicatorEl.getBoundingClientRect().width;
-                            indicatorParentEl.style.display = "";
+            if (this.snippetVersion >= 2) {
+                this.btnEls = this.indicatorEl.querySelectorAll("[data-bs-slide-to]");
```

Comment:
I find it clearer to call it `this.indicatorEls` (`btnEls`is too vague) => the plural distinguishes it from `indicatorEl`, and at worst, we could rename it `indicatorsWrapperEl`